src/main/java/me/yukun99/ip/core/Parser.java

```diff
@@ -0,0 +1,133 @@
+package me.yukun99.ip.core;
+
+import me.yukun99.ip.HelpBot;
+import me.yukun99.ip.commands.AddCommand;
+import me.yukun99.ip.commands.Command;
+import me.yukun99.ip.commands.DeleteCommand;
+import me.yukun99.ip.commands.DoneCommand;
+import me.yukun99.ip.commands.ExitCommand;
+import me.yukun99.ip.commands.FindCommand;
+import me.yukun99.ip.commands.HelpCommand;
+import me.yukun99.ip.commands.ListCommand;
+import me.yukun99.ip.commands.UpdateCommand;
+import me.yukun99.ip.exceptions.HelpBotIllegalArgumentException;
+import me.yukun99.ip.exceptions.HelpBotInvalidCommandException;
+
+import java.util.Scanner;
+
+import static me.yukun99.ip.tasks.Task.Type;
+
+/**
+ * Parser for HelpBot commands.
+ */
+public class Parser {
+	// Current instance of HelpBot.
+	private final HelpBot helpBot;
+	// Scanner used to scan for user inputs.
+	private final Scanner scanner;
+	// TaskList instance from the current instance of HelpBot.
+	private final TaskList taskList;
+	// Ui instance from the current instance of HelpBot.
+	private final Ui ui;
+	// Storage instance from the current instance of HelpBot.
+	private final Storage storage;
+
+	/**
+	 * Constructor for a Parser instance.
+	 *
+	 * @param helpBot HelpBot instance that this instance of Parser belongs to.
+	 * @param scanner Scanner used for user input scanning.
+	 * @param taskList TaskList instance from the current instance of HelpBot.
+	 * @param ui Ui instance from the current instance of HelpBot.
+	 * @param storage Storage instance from the current instance of HelpBot.
+	 */
+	public Parser(HelpBot helpBot, Scanner scanner, TaskList taskList, Ui ui, Storage storage) {
+		this.helpBot = helpBot;
+		this.scanner = scanner;
+		this.taskList = taskList;
+		this.ui = ui;
+		this.storage = storage;
+	}
+
+	/**
+	 * Parses the next command sent by the user.
+	 *
+	 * @return Next command sent to the HelpBot.
+	 * @throws HelpBotIllegalArgumentException If arguments provided for the commands are missing or wrong.
+	 * @throws HelpBotInvalidCommandException If command sent by the user is invalid.
+	 */
+	public final Command parseCommand() throws HelpBotIllegalArgumentException, HelpBotInvalidCommandException {
+		String message = scanner.nextLine();
+		return parseCommand(message);
+	}
+
+	private Command parseCommand(String message)
+			throws HelpBotInvalidCommandException, HelpBotIllegalArgumentException {
+		Command command = null;
+		if (message.equals("list")) {
+			command = new ListCommand(null, taskList, ui);
+		}
+		if (message.startsWith("list ")) {
```

Comment:
Maybe the if statements can be joined to form a large if-else if-else block?

Reply:
I think this one should be fine!